### PR TITLE
Refactor: Raise 404 for draft events with proper permission

### LINKED ISSUE #4193: Refactor: Raise 404 for draft events with proper permission
**Current behavior:**

Endpoints like /events/id/something do not thrown 404 if event is draft and user is not logged in or does not have the proper permissions.
**Expected behavior:**

 Refactor all such APIs to respect the dev-handbook

### PR DESCRIPTION
**Pull request by shubham-padia**
Fixes #4193

### CODE REVIEW

File: app/api/helpers/query.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from flask import request
+from flask_rest_jsonapi.exceptions import ObjectNotFound
+
+from app.api.helpers.db import safe_query
+from app.models.event import Event
+from app.api.helpers.permission_manager import has_access
+
+
+def event_query(self, query_, view_kwargs, event_id, event_identifier, permission):
```
Comment (poush) [inline app/api/helpers/query.py]: is it will be good if we provide default values as well to event_id and  event_identifier ?

Comment (niranjan94) [inline app/api/helpers/query.py]: Good job moving this to a common place ! 🎉

Comment (shubham-padia) [inline app/api/helpers/query.py]: Thanks for pointing it out @poush , the function call length reduced because of that :smile:

**Comment by shubham-padia** [comment]
@fossasia/open-event-server-devs Please review.

File: app/api/access_codes.py [hunk 2]
```diff
@@ -107,12 +108,7 @@ def query(self, view_kwargs):
         :return:
         """
         query_ = self.session.query(AccessCode)
```
Comment (SaptakS) [inline app/api/access_codes.py]: Why have this line if the query_ variable is changed in the very next line?

File: app/api/helpers/jwt.py [hunk 3]
```diff
@@ -32,3 +34,18 @@ def jwt_identity(payload):
     return User.query.get(payload['identity'])
 
 
+def get_identity():
+    """
+    To be used only if identity for expired tokens is required, otherwise use current_identity from flask_jwt
+    :return:
+    """
+    token_second_sengment = _default_request_handler().split('.')[1]
```
Comment (SaptakS) [inline app/api/helpers/jwt.py]: Typo: segment

File: app/api/helpers/query.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    :param event_identifier: String representing event_identifier in the view_kwargs
+    :param query_: Query object
+    :param view_kwargs: view_kwargs from the API
+    :param permission: the name of the permission to be applied as a string. Default: is_coorganizer
+    :return:
+    """
+    if view_kwargs.get(event_id):
+        event = safe_query(self, Event, 'id', view_kwargs[event_id], event_id)
+        if event.state != 'published':
+            if 'Authorization' in request.headers and has_access(permission, event_id=event.id):
```
Comment (SaptakS) [inline app/api/helpers/query.py]: I think you can reduce the number of nested if here. It is advisable to reduce the number of nested if to increase readability of code.

Comment (shubham-padia) [inline app/api/helpers/query.py]: IMO this looks good, if you have a better alternative i'll be glad to replace it :)

File: app/api/access_codes.py (hunk 2 above)
Comment (shubham-padia) [inline app/api/access_codes.py]: This is the base query, the join will be made by the function to this query.

File: app/api/helpers/jwt.py (hunk 3 above)
Comment (shubham-padia) [inline app/api/helpers/jwt.py]: Thanks, resolved

**Comment by shubham-padia** [comment]
@SaptakS please review.

File: app/api/helpers/query.py (hunk 4 above)
Comment (SaptakS) [inline app/api/helpers/query.py]:
```python
    if view_kwargs.get(event_id):
        event = safe_query(self, Event, 'id', view_kwargs[event_id], event_id)
        if event.state != 'published' and ('Authorization' not in request.headers or \
            not has_access(permission, event_id=event.id)):
            raise ObjectNotFound({'parameter': event_id},
                                     "Event: {} not found".format(view_kwargs[event_id]))
        query_ = query_.join(Event).filter(Event.id == event.id)
```

Isn't this better?

Comment (shubham-padia) [inline app/api/helpers/query.py]: @SaptakS Yep, Thanks. Resolved.

**Comment by shubham-padia** [comment]
@SaptakS please review.

**Comment by SaptakS** [comment]
LGTM. Merging. 🍪

**Pull request merged**